lib/igniter.ex

```diff
@@ -1123,48 +1123,62 @@ defmodule Igniter do
           source -> source
         end
 
-      if Rewrite.Source.from?(source, :string) do
-        content_lines =
-          source
-          |> Rewrite.Source.get(:content)
-          |> String.split("\n")
-          |> Enum.with_index()
-
-        space_padding =
-          content_lines
-          |> Enum.map(&elem(&1, 1))
-          |> Enum.max()
-          |> to_string()
-          |> String.length()
-
-        diffish_looking_text =
-          Enum.map_join(content_lines, "\n", fn {line, line_number_minus_one} ->
-            line_number = line_number_minus_one + 1
-
-            "#{String.pad_trailing(to_string(line_number), space_padding)} #{color(IO.ANSI.yellow(), color?)}| #{color(IO.ANSI.green(), color?)}#{line}#{color(IO.ANSI.reset(), color?)}"
-          end)
+      cond do
+        Rewrite.Source.from?(source, :string) &&
+            String.valid?(Rewrite.Source.get(source, :content)) ->
+          content_lines =
+            source
+            |> Rewrite.Source.get(:content)
+            |> String.split("\n")
+            |> Enum.with_index()
+
+          space_padding =
+            content_lines
+            |> Enum.map(&elem(&1, 1))
+            |> Enum.max()
+            |> to_string()
+            |> String.length()
+
+          diffish_looking_text =
+            Enum.map_join(content_lines, "\n", fn {line, line_number_minus_one} ->
+              line_number = line_number_minus_one + 1
+
+              "#{String.pad_trailing(to_string(line_number), space_padding)} #{color(IO.ANSI.yellow(), color?)}| #{color(IO.ANSI.green(), color?)}#{line}#{color(IO.ANSI.reset(), color?)}"
+            end)
 
-        if String.trim(diffish_looking_text) != "" do
-          """
-          Create: #{Rewrite.Source.get(source, :path)}
+          if String.trim(diffish_looking_text) != "" do
+            """
+            Create: #{Rewrite.Source.get(source, :path)}
 
-          #{diffish_looking_text}
-          """
-        else
-          ""
-        end
-      else
-        diff = Rewrite.Source.diff(source, color: color?) |> IO.iodata_to_binary()
+            #{diffish_looking_text}
+            """
+          else
+            ""
+          end
+
+        String.valid?(Rewrite.Source.get(source, :content)) ->
+          diff = Rewrite.Source.diff(source, color: color?) |> IO.iodata_to_binary()
+
+          if String.trim(diff) != "" do
+            """
+            Update: #{Rewrite.Source.get(source, :path)}
 
-        if String.trim(diff) != "" do
+            #{diff}
+            """
+          else
+            ""
+          end
+
+        !String.valid?(Rewrite.Source.get(source, :content)) ->
```

Comment:
Some files do not have a valid string content, for eg binary files like `favicon.ico` but I think we still need to display that such file will be created. Might need to improve this code to detect updates as well.

Reply:
We should probably add it in this or to avoid confusion. Can use `changed?`